python/plugins/processing/tools/postgis.py

```diff
@@ -138,41 +163,54 @@ def _quote(self, ident):
 
 class GeoDB:
 
+    @classmethod
+    def from_name(cls, name):
+        uri = uri_from_name(name)
+        return cls(uri=uri)
+
     def __init__(self, host=None, port=None, dbname=None, user=None,
-                 passwd=None):
+                 passwd=None, service=None, uri=None):
         # Regular expression for identifiers without need to quote them
         self.re_ident_ok = re.compile(r"^\w+$")
 
-        self.host = host
-        self.port = port
-        self.dbname = dbname
-        self.user = user
-        self.passwd = passwd
-
-        if self.dbname == '' or self.dbname is None:
-            self.dbname = self.user
-
-        try:
-            self.con = psycopg2.connect(self.con_info())
-        except psycopg2.OperationalError as e:
-            raise DbError(unicode(e))
+        if uri:
+            self.uri = uri
+        else:
+            self.uri = QgsDataSourceURI()
+            if service:
+                self.uri.setConnection(service, dbname, user, passwd)
+            else:
+                self.uri.setConnection(host, port, dbname, user, passwd)
+
+        conninfo = self.uri.connectionInfo(False)
+        for i in range(3):
+            try:
+                self.con = psycopg2.connect(self.uri.connectionInfo(True))
```

Comment:
Hi @arnaud-morvan Just a note... this could work only in case of basic auth. In general would be better to use DbManager interface, btw in this case you would manage PKI connection setting credentials.
You can be inspired by the same as in DbManager.
As you wrote this piece of code it will work only with db connection with basic auth.

In other parts of this PR you do expansion "uri.connectionInfo(True)" without checking if URI has user and pwd inside or credentials are stored in QgsAuthManager with different auth system other than Basic Auth.

I would suggest to
1) check if uri params has authcfg
2) if so, ask to QgsAuthManager type of credentials
3) if not basic auth => notify i't not uspported by this code... of modify the code to support it